library/general/replicalimits/template.yaml

```diff
@@ -0,0 +1,47 @@
+apiVersion: templates.gatekeeper.sh/v1beta1
+kind: ConstraintTemplate
+metadata:
+  name: k8sreplicalimits
+  description: Requires a number of replicas to be set for a deployment between a min and max value.
+spec:
+  crd:
+    spec:
+      names:
+        kind: K8sReplicaLimits
+      validation:
+        # Schema for the `parameters` field
+        openAPIV3Schema:
+          properties:
+            ranges:
+              type: array
+              items:
+                type: object
+                properties:
+                  min_replicas:
+                    type: integer
+                  max_replicas:
+                    type: integer
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package k8sreplicalimits
+
+        deployment_name = input.review.object.metadata.name
+
+        violation[{"msg": msg}] {
+          spec := input.review.object.spec
+          not input_replica_limit(spec)
+          msg := sprintf("The provided number of replicas is not allowed for deployment: %v. Allowed ranges: %v", [deployment_name, input.parameters])
```

Comment:
nit: since we are not using `deployment_name` anywhere else, we can just specify `input.review.object.metadata.name` here